### ISSUE TITLE
HttpClient should be used through IHttpClientFactory 

### ISSUE THREAD

**Issue by jamiehankins** [ShipEngine/shipengine-dotnet#115]
This article [You're using HttpClient wrong and it's destabilizing your software](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/) describes the problem. Their solution is to share a single instance of `HttpClient`.

A better solution is to implement `IHttpClientFactory`. Your `ShipEngineClient` constructor could take an instance of this. That would make it easy to adapt to applications that use dependency injection.

Here's the [documentation for `IHttpClientFactory`](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests).

**Comment by AnneOReilly** [comment]
Thanks!  We are reviewing this - it sounds like there were some changes earlier this year that should have resolved this issue, but we're going to double check and clear up anything we missed.  We'll close this when that is merged in.

**Comment by joshuaflanagan** [comment]
You are absolutely right that this _was_ an issue, until our 2.2.1 release in June 2024.

We added the [new constructor](https://github.com/ShipEngine/shipengine-dotnet/blob/ce007a668e01c031b13ed483081860e49fa13d0e/ShipEngineSDK/ShipEngine.cs#L222) so that you could pass in an `HttpClient` instance that you manage yourself or obtain from `IHttpClientFactory`. We did not remove the existing constructors that create their own instance as we did not want to break compatibility. This is something we are considering for the 3.x release.

We also added [a convenience method](https://github.com/ShipEngine/shipengine-dotnet/blob/ce007a668e01c031b13ed483081860e49fa13d0e/ShipEngineSDK/ShipEngine.cs#L169) for registering a [typed client](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests#how-to-use-typed-clients-with-ihttpclientfactory) that leverages `IHttpClientFactory` under the hood.

**Comment by jamiehankins** [comment]
Thanks, looks good.
